bindings/node/native/src/encoding.rs

```diff
@@ -182,53 +206,92 @@ declare_types! {
             }
         }
 
-        method tokenToChars(mut cx) {
-            // tokenToChars(token: number): [number, number] | undefined
+        method tokenToSequence(mut cx) {
+            // tokenToSequence(token: number): number | undefined
 
-            let token = cx.argument::<JsNumber>(0)?.value() as usize;
+            let token = cx.extract::<usize>(0)?;
 
             let this = cx.this();
             let guard = cx.lock();
 
             let res = this.borrow(&guard)
                 .encoding.as_ref().expect("Uninitialized Encoding")
-                .token_to_chars(token);
+                .token_to_sequence(token);
 
-            if let Some(offsets) = res {
-                Ok(neon_serde::to_value(&mut cx, &offsets)?)
+            if let Some(seq) = res {
+                Ok(neon_serde::to_value(&mut cx, &seq)?)
+            } else {
+                Ok(cx.undefined().upcast())
+            }
+        }
+
+        method tokenToChars(mut cx) {
+            // tokenToChars(token: number): [number, number] [number, [number, number]] | undefined
+
+            let token = cx.extract::<usize>(0)?;
+
+            let this = cx.this();
+            let guard = cx.lock();
+
+            let (res, n_seq) = {
+                let borrowed = this.borrow(&guard);
+                let encoding = borrowed.encoding.as_ref().expect("Uninitialized Encoding");
+
+                let res = encoding.token_to_chars(token);
+                let n_seq = encoding.n_sequences();
+                (res, n_seq)
+            };
+
+            if let Some((seq_id, offsets)) = res {
+                if n_seq > 1 {
+                    Ok(neon_serde::to_value(&mut cx, &(seq_id, offsets))?)
+                } else {
+                    Ok(neon_serde::to_value(&mut cx, &offsets)?)
+                }
             } else {
                 Ok(cx.undefined().upcast())
             }
         }
 
         method tokenToWord(mut cx) {
-            // tokenToWord(token: number): number | undefined
+            // tokenToWord(token: number): number | [number, number] | undefined
 
             let token = cx.argument::<JsNumber>(0)?.value() as usize;
 
             let this = cx.this();
             let guard = cx.lock();
-            let index = this.borrow(&guard)
-                .encoding.as_ref().expect("Uninitialized Encoding")
-                .token_to_word(token);
 
-            if let Some(index) = index {
-                Ok(cx.number(index as f64).upcast())
+            let (res, n_seq) = {
+                let borrowed = this.borrow(&guard);
+                let encoding = borrowed.encoding.as_ref().expect("Uninitialized Encoding");
+
+                let res = encoding.token_to_word(token);
+                let n_seq = encoding.n_sequences();
+                (res, n_seq)
+            };
+
+            if let Some((seq_id, index)) = res {
+                if n_seq > 1 {
+                    Ok(neon_serde::to_value(&mut cx, &(seq_id, index))?)
```

Comment:
TBH variant return types always give me shivers. It's usually so much better to have 2 different functions instead. Saves so much headaches for users....

Reply:
As discussed, we'll remove the output with the sequence id, and will just keep the classic one. The user can call `token_to_sequence` when needed to determine it.